tremor-script/src/ast/helper.rs

```diff
@@ -28,14 +28,40 @@ use crate::{
 };
 use beef::Cow;
 use halfbrown::HashMap;
-use std::{collections::BTreeSet, mem};
+use std::{collections::BTreeSet, fmt::Display, mem};
 
 /// ordered collection of warnings
 pub type Warnings = std::collections::BTreeSet<Warning>;
 
+/// Class of warning that gives additional insight into the warning's intent
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Copy)]
+pub enum WarningClass {
```

Comment:
what about moving the warning struct etc. to a new module `warning.rs` and naming this only `Category`?

Reply:
What about adding a `Deprecation` variant?

And should we remove `std::array::concatenate` and `std::record::combine` if we have constructs that work better? Or should we keep them, as they are nice pointers to the cool constructs we have?

Reply:
1) warning module 👍 good call, will do

2) deprication warning, yes I had it, but we don't have any so it'll yell at us for dead code, we can add them once we get them :)

3) I was thinking about it and didn't since eventually we want higher order functions and function arguments where we'll need to concatenate and combine as functions